### PR TITLE
Simplified regex in removeJsonQuotes

### PR DESCRIPTION
**Pull request by jdsteinhauser**
Resolves Issue #415. I'm not satisfied with how I had to test it, since Jest wasn't happy with any way I tried to import `src/lib/fetch.[jt]s` . If an RFC is needed, I'll gladly write one but I was assuming this was small and straightforward enough to not require one.

- cleaned up the tests
- added some tests
- made sure it works with safari

### CODE REVIEW

**Comment by shimkiv** [comment]
Seems to work because now it reveals [another issue](https://stackoverflow.com/a/65099653) (no stacktrace available):

image

File: src/lib/fetch.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  let cleaned = JSON.stringify(JSON.parse(json), null, 2);
+  return cleaned.replace(/\"(\S+)\"\s*:/gm, "$1 :");
+}
+
+// function removeJsonQuotes(json: string) {
+//   let cleaned = JSON.stringify(JSON.parse(json), null, 2);
+//   return cleaned.replace(/\"([\w-]+)\"\s*:/gm, "$1 :");
+// }
+
+describe('fetch helper functions', () => {
```
Comment (mitschabaude) [inline src/lib/fetch.test.ts:16]: can we also test nested objects + arrays, e.g.
```json
{
  "array": [
    {
      "data": "string with \"escaped quotes\": 1"
    }
  ]
}
```

**Review by mitschabaude — COMMENTED** [review]
Very nice! One more test and I'm happy :D

The web workers thing is a bummer, but we'll manage to work around that as well

**Comment by mitschabaude** [comment]
@jdsteinhauser re: testing. Our Jest setup currently doesn't manage to build the whole project, so right now with Jest we can only test stuff that is exported from snarkyjs (all the other Jest test import `"snarkyjs"` which resolves to the build output in `/dist/node`).

We recently introduced a way to test internal stuff as well -- you can see it demonstrated in `src/lib/hash-inputs.unit-test.ts`.
-> Just make a new file with the ending `unit-test.ts`, and it will automatically be built and run as part of `npm run test:unit`.
You can't use `describe` and `it` in these files, but you _can_ use `expect` (which is just a utility function that throws an error if a test fails), so you get the same testing capabilities, just with less fancy console output

**Comment by bkase** [comment]
@jsteinhauser could you rebase the PR off of `main` instead?

**Comment by Trivo25** [comment]
- cleaned up the tests
- added some tests
- made sure it works with safari

**Review by mitschabaude — APPROVED** [review]
awesome!

**Comment by jdsteinhauser** [comment]
Oh my goodness, thanks for finishing this one out @Trivo25 !

**Comment by Trivo25** [comment]
> Oh my goodness, thanks for finishing this one out @Trivo25 !

Sorry it took so long :D

**Pull request merged**